server/src/main/java/com/hedera/block/server/persistence/storage/PersistenceStorageConfig.java

```diff
@@ -56,11 +62,28 @@ public record PersistenceStorageConfig(
      */
     public PersistenceStorageConfig {
         Objects.requireNonNull(type);
+        Objects.requireNonNull(compression);
+        verifyCompressionLevel(compression, compressionLevel);
```

Comment:
Simplification:
```suggestion
        verifyCompressionLevel(Objects.requireNonNull(compression), compressionLevel);
```
OR (see below)
```suggestion
        compression.verifyCompressionLevel(compressionLevel);
```

Reply:
✅ Done.